build.rs

```diff
@@ -3,28 +3,81 @@ use std::{env, fs, path::Path};
 fn main() -> anyhow::Result<()> {
     println!("cargo:rerun-if-env-changed=./src");
 
+    // Check that required tools are available
+    check_tool_available("curl")?;
+    check_tool_available("tar")?;
+
     let output = env::var("OUT_DIR").unwrap();
     let commit_hash = "b7d97d5f3f8f897b88872b6935e4c996b955bc1f";
     let source_dir = Path::new(&output).join("libyuv-source");
 
-    // Clone the repository to the specific commit
+    // Download the repository as a tarball to avoid git authentication issues
     if source_dir.exists() {
         fs::remove_dir_all(&source_dir)?;
     }
 
-    println!("cargo:warning=Cloning libyuv at commit {}", commit_hash);
-    let repo = git2::Repository::clone("https://github.com/lemenkov/libyuv", &source_dir)?;
-    let oid = git2::Oid::from_str(commit_hash)?;
-    let commit = repo.find_commit(oid)?;
-    repo.checkout_tree(commit.as_object(), None)?;
-    repo.set_head_detached(oid)?;
+    println!("cargo:warning=Downloading libyuv at commit {}", commit_hash);
+
+    // Download tarball from GitHub
+    let tarball_url = format!(
+        "https://github.com/lemenkov/libyuv/archive/{}.tar.gz",
+        commit_hash
+    );
+    let tarball_path = Path::new(&output).join("libyuv.tar.gz");
+
+    // Download the tarball
+    let response = std::process::Command::new("curl")
+        .args(&["-L", "-o", tarball_path.to_str().unwrap(), &tarball_url])
+        .output()?;
```

Comment:
Shelling out.... hmmm...

What if there was a simple sync Rust http client for this kind of stuff? :)